src/Moq.EntityFrameworkCore.Examples/Users/UsersService.cs

```diff
@@ -41,5 +41,20 @@ public async Task<User> FindOneUserAsync(Expression<Func<User, bool>> predicate)
         {
             return await this.usersContext.Set<User>().FirstOrDefaultAsync(predicate);
         }
+
+        public async Task<IList<User>> ChangeSetInSequence()
```

Comment:
I would change this example. This is a bit complicated and not natural one - you need to think a bit to understand it.

You can use one of the previous methods and setup as you proposed. And execute method in Act two times - e.g. find disabled users. First time list should be empty. And second time according to users in the list.

It will be easier to understand.

Reply:
@MichalJankowskii looking at this now, I could use a little help understanding better what you expect here. The idea is the unit test calls a method that takes two actions on the same dbset. I ran into exactly this case in a real product where I needed to test a service that made changes to a dbset several times, and the set needed to be mocked in sequence to test properly.

I believe what you're proposing is changing the set in the test itself, which doesn't seem representative of what you use SetupSequence for - you're expecting the set to change in the method you're testing. Please help me understand what you're proposing and whether I'm misunderstanding your suggestion, thanks!